lib/ex_plasma/transaction.ex

```diff
@@ -242,4 +245,16 @@ defmodule ExPlasma.Transaction do
     sigs = Enum.map(keys, fn key -> ExPlasma.Encoding.signature_digest(eip712_hash, key) end)
     %{transaction | sigs: sigs}
   end
+
+  # Builds list of utxos and propogates error tuples up the stack.
```

Comment:
maybe you'd enjoy this implementation more (with simple recursive calls):
```suggestion
  # Builds list of utxos and propogates error tuples up the stack.
  # Builds list of utxos and propogates error tuples up the stack.
  defp build_utxos(utxos), do: build_utxos(utxos, [])
  defp build_utxos([], acc), do: {:ok, acc}

  defp build_utxos([utxo | utxos], acc) do
    case Utxo.new(utxo) do
      {:ok, utxo} ->
        build_utxos(utxos, [utxo | acc])

      {:error, reason} ->
        {:error, reason}
    end
  end
```